### PR TITLE
Add Java 17 to build/test matrix

### PR DESCRIPTION
**Pull request by basil**

### CODE REVIEW

**Comment by basil** [comment]
Used my admin powers to replay the build with the `Jenkinsfile` changes here: https://ci.jenkins.io/job/Plugins/job/active-directory-plugin/job/PR-168/2/

File: Jenkinsfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 buildPlugin(configurations: [
-    [platform: 'linux', jdk: '11'],
-    [platform: 'windows', jdk: '11'],
+    [platform: 'linux', jdk: 17],
+    [platform: 'windows', jdk: 11],
```
Comment (basil) [inline Jenkinsfile:3]: Following the archetype.

File: pom.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>4.50</version>
+    <version>4.63</version>
```
Comment (basil) [inline pom.xml:7]: Latest at the time of this writing.

File: pom.xml [hunk 3]
```diff
@@ -37,15 +37,15 @@
     <revision>2.31</revision>
     <changelist>-SNAPSHOT</changelist>
     <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
-    <jenkins.version>2.332.4</jenkins.version>
+    <jenkins.version>2.361.4</jenkins.version>
```
Comment (basil) [inline pom.xml:40]: Minimum version supported by the latest plugin parent POM.

File: pom.xml [hunk 4]
```diff
@@ -37,15 +37,15 @@
...
   </properties>
 
   <dependencyManagement>
     <dependencies>
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
-        <artifactId>bom-2.332.x</artifactId>
-        <version>1678.vc1feb_6a_3c0f1</version>
+        <artifactId>bom-2.361.x</artifactId>
+        <version>2081.v85885a_d2e5c5</version>
```
Comment (basil) [inline pom.xml:48]: Latest at the time of this writing.

File: src/main/java/hudson/plugins/active_directory/ActiveDirectorySecurityRealm.java [hunk 5]
```diff
@@ -45,7 +45,6 @@
 import org.acegisecurity.providers.UsernamePasswordAuthenticationToken;
 import org.acegisecurity.userdetails.UserDetails;
 import org.acegisecurity.userdetails.UsernameNotFoundException;
-import org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement;
```
Comment (basil) [inline src/main/java/hudson/plugins/active_directory/ActiveDirectorySecurityRealm.java:48]: Animal Sniffer has been removed from the latest plugin parent POM.

File: src/test/java/hudson/plugins/active_directory/ActiveDirectoryJCasCCompatibilityTest.java [hunk 6]
```diff
@@ -10,6 +11,7 @@
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
+@Ignore("setting environment variables is not supported in Java 17")
```
Comment (basil) [inline src/test/java/hudson/plugins/active_directory/ActiveDirectoryJCasCCompatibilityTest.java:14]: https://github.com/stefanbirkner/system-rules/issues/64#issuecomment-401944327

File: src/test/resources/fixture/Dockerfile [hunk 7]
```diff
@@ -1,14 +1,5 @@
...
-RUN apk add -X https://dl-cdn.alpinelinux.org/alpine/v3.16/main -u alpine-keys --allow-untrusted
-# Workaround for  https://github.com/AdoptOpenJDK/openjdk-docker/issues/75
-RUN apk add --no-cache fontconfig ttf-dejavu openjdk11 bash tini bind-tools
-RUN apk add aufs-util --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing
-RUN apk add maven --repository http://dl-cdn.alpinelinux.org/alpine/edge/community
-# Workaround for  https://github.com/AdoptOpenJDK/openjdk-docker/issues/75
-RUN ln -s /usr/lib/libfontconfig.so.1 /usr/lib/libfontconfig.so && \
-    ln -s /lib/libuuid.so.1 /usr/lib/libuuid.so.1 && \
-    ln -s /lib/libc.musl-x86_64.so.1 /usr/lib/libc.musl-x86_64.so.1
+FROM docker:dind
```
Comment (basil) [inline src/test/resources/fixture/Dockerfile:1]: The `stable-dind` image is no longer available or maintained.

File: src/test/resources/fixture/Dockerfile [hunk 8]
```diff
@@ -1,14 +1,5 @@
...
+RUN apk add --no-cache fontconfig ttf-dejavu openjdk11 bash tini bind-tools maven
```
Comment (basil) [inline src/test/resources/fixture/Dockerfile:2]: Previous workarounds are no longer necessary in a modern image.

**Review by fbelzunc — APPROVED** [review]

**Comment by basil** [comment]
Closing in favor of #170.

**Pull request closed without merging**